src/cpp/fastdds/builtin/type_lookup_service/detail/rpc_types.idl

```diff
@@ -0,0 +1,83 @@
+//#ifdef BASIC
```

Comment:
I suggest adding a comment indicating this have been modified with respect to the online IDL. Something in the lines: `Pending support for preprocessor definitions`. I suggest adding this to other lines commented for this reason.